File: ITapply/ITapply.Models/SearchObjects/UserSearchObject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ITapply.Models.SearchObjects
{
    public class UserSearchObject : BaseSearchObject
    {
        public string Email { get; set; } = string.Empty;
        public DateTime? RegistrationDate { get; set; }
        public bool? IsActive { get; set; }
    }
}

File: ITapply/ITapply.UI/itapply_mobile/lib/models/search_objects/user_search_object.dart
import 'package:itapply_desktop/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_search_object.g.dart';

@JsonSerializable()
class UserSearchObject extends BaseSearchObject {
  String? Email;
  DateTime? RegistrationDate;
  bool? IsActive;

  UserSearchObject({
    this.Email,
    this.RegistrationDate,
    this.IsActive,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$UserSearchObjectToJson(this);
}